src/global_distrib/mod_global_distrib_hosts_refresher.erl

```diff
@@ -0,0 +1,154 @@
+%% limitations under the License.
+%%==============================================================================
+
+
+-module(mod_global_distrib_hosts_refresher).
+-author("dominik.stanaszek@erlang-solutions.com").
+
+-behavior(gen_server).
+-behavior(gen_mod).
+-include("mongoose.hrl").
+
+%% API
+-export([start_link/1]).
+
+%% gen_server API
+-export([init/1, handle_call/3, handle_cast/2, handle_info/2, terminate/2]).
+
+%% gen_mod API
+-export([deps/2, start/2, stop/1]).
+
+
+%%--------------------------------------------------------------------
+%% API
+%%--------------------------------------------------------------------
+
+-spec start_link(Milliseconds :: non_neg_integer()) -> pid() | term().
+start_link(RefreshInterval) ->
+    gen_server:start_link({local, ?MODULE}, ?MODULE, [RefreshInterval], []).
+
+
+%%--------------------------------------------------------------------
+%% gen_mod callbacks
+%%--------------------------------------------------------------------
+
+-spec start(Host :: ejabberd:lserver(), Opts :: proplists:proplist()) -> any().
+start(Host, Opts) ->
+    mod_global_distrib_utils:start(?MODULE, Host, Opts, fun start/0).
+
+-spec stop(Host :: ejabberd:lserver()) -> any().
+stop(Host) ->
+    mod_global_distrib_utils:stop(?MODULE, Host, fun stop/0).
+
+-spec deps(Host :: ejabberd:server(), Opts :: proplists:proplist()) -> gen_mod:deps_list().
+deps(Host, Opts) ->
+    mod_global_distrib_utils:deps(?MODULE, Host, Opts, fun deps/1).
+
+%%--------------------------------------------------------------------
+%% gen_server callbacks
+%%--------------------------------------------------------------------
+
+init([RefreshInterval]) ->
+    ?DEBUG("refresher starting with interval ~p~n", [RefreshInterval]),
+    schedule_refresh(RefreshInterval),
+    {ok, RefreshInterval}.
+
+handle_call(Request, From, State) ->
+    ?ERROR_MSG("issue=unknown_call request=~p from=~p", [Request, From]),
+    {reply, {error, unknown_request}, State}.
+
+handle_cast(Request, State) ->
+    ?ERROR_MSG("issue=unknown_cast request=~p", [Request]),
+    {noreply, State}.
+
+handle_info(refresh, Interval) ->
+    refresh(),
+    schedule_refresh(Interval),
+    {noreply, Interval}.
+
+terminate(normal, _State) ->
+    ?INFO_MSG("mod_global_distrib_refresher has terminated with reason: normal", []).
+
+
+%%--------------------------------------------------------------------
+%% Helper functions
+%%--------------------------------------------------------------------
+
+
+-spec deps(Opts :: proplists:proplist()) -> gen_mod:deps_list().
+deps(Opts) ->
+    [{mod_global_distrib_mapping, Opts, hard}].
+
+-spec start() -> any().
+start() ->
+    start_outgoing_conns_sup(),
+    Interval = mod_global_distrib_utils:opt(?MODULE, hosts_refresh_interval, default_refresh_interval()),
+    Child = #{
+      id => mod_global_distrib_hosts_refresher,
+      start => {?MODULE, start_link, [Interval]},
+      restart => transient,
```

Comment:
Why `transient`?

Reply:
Well, we want it to be restarted in case of error, don't we?

Reply:
The full question would be "why `transient` as opposed to `permanent`?".